x/protocolpool/types/types.go

```diff
@@ -0,0 +1,42 @@
+package types
+
+import (
+	"errors"
+
+	"cosmossdk.io/math"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+// PercentageCoinMul multiplies each coin in an sdk.Coins struct by the given percentage and returns the new
+// value.
+//
+// When performing multiplication, the resulting values are truncated to an sdk.Int.
+func PercentageCoinMul(percentage math.LegacyDec, coins sdk.Coins) sdk.Coins {
+	ret := sdk.NewCoins()
+
+	for _, denom := range coins.Denoms() {
+		am := sdk.NewCoin(denom, percentage.MulInt(coins.AmountOf(denom)).TruncateInt())
+		ret = ret.Add(am)
+	}
+
+	return ret
+}
```

Comment:
this seems to only be used in the keeper package. should we move there?

Reply:
We can, I try to keep any functions that don't use `keeper` or `ctx` in the `types` package for organization

Reply:
i think its better to organize functionality where its used tbh. otherwise you end up with a `types` package that is muddy in what it provides, further obfuscating its purpose. 

additionally, we can make this function private if we move to the keeper package, and reduce the amount of functionality exposed from `types` - simplifying the API